src/main/java/com/autotune/analyzer/utils/ExperimentResultValidation.java

```diff
@@ -54,7 +56,17 @@ public void validate(List<ExperimentResultData> experimentResultDataList) {
                             resultData.setValidationResultData(new ValidationResultData(false, errorMsg));
                             break;
                         }
-                        proceed = true;
+                        // Validate Performance Profile data
+                        PerfProfileInterface perfProfileInterface = new ResourceOptimizationOpenShift();
```

Comment:
As mentioned the performance profile should be picked based on what is stored in the `KruizeObject` and not hard coded. Also please use `perfProfileImpl` instead of `perfProfileInterface`

Reply:
Updated